compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TileAndDecomposeAttention.cpp

```diff
@@ -505,6 +535,30 @@ void decomposeTiledAttention(IREE::LinalgExt::AttentionOp tiledAttnOp,
       tileSize ? rewriter.getIndexAttr(tileSize.value()) : sequenceTileLength;
 
   Type elementType = tiledAttnOp.getQueryType().getElementType();
+
+  assert(isa<FloatType>(elementType) &&
+         "Attention only works on floating types");
+  FloatType floatElementType = cast<FloatType>(elementType);
+
+  // Since we use exp2 for attention instead of the original exp, we have to
+  // multiply the scale by log2(e) = 1.44269504089. We use exp2 instead of exp
+  // as most gpus have better support for exp2.
+  Value scale = tiledAttnOp.getScale();
+  Value log2e = rewriter.create<arith::ConstantFloatOp>(
+      loc, APFloat(1.44269504089), floatElementType);
+  scale = rewriter.create<arith::MulFOp>(loc, scale, log2e);
+
+  // In the original algorithm, the scaling is done after the softmax:
+  //        softmax(Q @ K.T / scale) @ V
+  //
+  // But, it is mathematically equivalent to do it on Q first and then multiply
+  // it by K.T. This just allows us to do the scaling once, instead of each
+  // iteratiion of the loop.
+  // (This was also verified on fp16 attention that it produces the same
+  // results).
+  querySlice = scaleQuery(querySlice, tiledAttnOp.getScale(), rewriter);
```

Comment:
shouldn't this use scale and not tiledAttnOp.getScale() ?

Reply:
fixed